vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java

```diff
@@ -0,0 +1,344 @@
+package com.vaadin.flow.component.combobox;
+
+import com.vaadin.flow.component.AbstractSinglePropertyField;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.Focusable;
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.Synchronize;
+import com.vaadin.flow.function.SerializableBiFunction;
+import com.vaadin.flow.component.combobox.events.CustomValueSetEvent;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Provides base functionality for combo box related components, such as
+ * {@link ComboBox}
+ *
+ * @param <TComponent>
+ *            Type of the component that extends from this class
+ * @param <TItem>
+ *            Type of individual items that are selectable in the combo box
+ * @param <TValue>
+ *            Type of the selection / value of the extending component
+ */
+public abstract class ComboBoxBase<TComponent extends ComboBoxBase<TComponent, TItem, TValue>, TItem, TValue>
+        extends AbstractSinglePropertyField<TComponent, TValue>
+        implements HasStyle, Focusable<TComponent> {
+
+    private static final String PROP_AUTO_OPEN_DISABLED = "autoOpenDisabled";
```

Comment:
nit: personally I don't see any reason for this specific property to be extracted into a constant.
IMO we can inline it for now, same as it's done for any other properties e.g. `placeholder` etc.

We might want to extract this into an interface e.g. `HasAutoOpen` later on.

Reply:
Agree, done